docs/content/docs/build/auth.md

```diff
@@ -0,0 +1,107 @@
+  "scopes": [ "doc:read", "doc:write", "summary:write" ],
+  "iat": 1599098963,
+  "exp": 1599098963,
+  "tenantId": "AzureFluidTenantId",
+  "ver": "1.0"
+}.[Signature]
+```
+
+Every request to FRS must be signed with a valid JWT. The FRS documentation contains additional details about [how to
+sign the token][1]. Fluid delegates the responsibility of creating and signing these tokens to a *token provider.*
+
+[1]: (https://github.com/MicrosoftDocs/azure-fluid-preview-pr/blob/main/azure-fluid-relay-preview-pr/articles/howtos/fluid-jwtoken.md#how-can-you-generate-an-azure-fluid-relay-token)
+
+{{% callout title="More information" %}}
+
+* [Introduction to JWTs](https://jwt.io/introduction)
+* [Payload claims in FRS](https://github.com/MicrosoftDocs/azure-fluid-preview-pr/blob/main/azure-fluid-relay-preview-pr/articles/howtos/fluid-jwtoken.md#payload-claims)
+* [Scopes in FRS](need a url)
+* [Signing requests](https://github.com/MicrosoftDocs/azure-fluid-preview-pr/blob/main/azure-fluid-relay-preview-pr/articles/howtos/fluid-jwtoken.md#how-can-you-generate-an-azure-fluid-relay-token)
+
+{{% /callout %}}
+
+## The token provider
+
+A token provider is responsible for creating and signing tokens that the Fluid runtime then uses to make requests to the
+service (FRS in this case). You are expected to provide your own token provider implementation for most scenarios.
+However, Fluid provides an `InsecureTokenProvider` that accepts your FRS tenant secret and returns signed tokens. This
+token provider is useful for testing, but in production scenarios you should use a more secure token provider.
+
+### A secure serverless token provider
+
+One option for building a secure token provider is to create a serverless Azure Function and expose it as a token
+provider. This enables you to store the FRS secret on a secure server. Your application calls the Function to generate
+tokens rather than signing them locally like the `InsecureTokenProvider` does.
+
+An example of such a function is available at <https://github.com/microsoft/FrsAzureFunctions>. There is a
+corresponding `FrsAzFunctionTokenProvider` for this Function in the `@fluid-experimental/frs-client` package.
+
+## Adding custom data to tokens
+
+Why would you do this? How does it work?
+
+## Connecting user auth to Fluid service auth
+
+You do this in your token provider. For example, you could make your Azure Function token provider authenticated. If an
+application tries to call the Function it would fail unless authenticated with your auth system. If you're using Azure
+Active Directory, for example, you might create an AAD application for your Azure Function, and tie it to your
+organization's auth system.
+
+In this case the user would sign into your application using AAD, through which you would obtain a token to use to call
+your Azure Function. The Azure Function itself behaves the same, but it's now only accessible to people who have also
+authenticated with AAD.
```

Comment:
```suggestion
authenticated with AAD.

Since the Azure Function is now your entrypoint into obtaining a valid token, only users who have properly authenticated to the Function will then be able to relay that token to FRS from their client application. This two-step approach allows you to use your own custom authentication process in conjunction with FRS.
```